variables.tf

```diff
@@ -1,6 +1,163 @@
+  type = object({
+    rule = object({             # (Required) Configuration block with Ownership Controls rules.
+      object_ownership = string # (Required) Object ownership. Valid values: BucketOwnerPreferred, ObjectWriter or BucketOwnerEnforced
+    })
+  })
+  default = null
+}
+
+variable "request_payment_configuration" {
+  description = "Request payment configuration for the S3 bucket"
+  type = object({
+    expected_bucket_owner = string # (Optional, Forces new resource) The account ID of the expected bucket owner.
+    payer                 = string # (Required) Specifies who pays for the download and request fees. Valid values: BucketOwner, Requester.
+  })
+  default = null
+}
+
+variable "cors_configuration" {
+  description = "List of cors rules the S3 bucket"
+  type = object({
+    expected_bucket_owner = string  # (Optional, Forces new resource) The account ID of the expected bucket owner.
+    cors_rules = set(object({       # (Required) Set of origins and methods (cross-origin access that you want to allow). You can configure up to 100 rules.
+      allowed_headers = set(string) # (Optional) Set of Headers that are specified in the Access-Control-Request-Headers header.
+      allowed_methods = set(string) # (Required) Set of HTTP methods that you allow the origin to execute. Valid values are GET, PUT, HEAD, POST, and DELETE.
+      allowed_origins = set(string) # (Required) Set of origins you want customers to be able to access the bucket from.
+      expose_headers  = set(string) # (Optional) Set of headers in the response that you want customers to be able to access from their applications (for example, from a JavaScript XMLHttpRequest object).
+      id              = string      # (Optional) Unique identifier for the rule. The value cannot be longer than 255 characters.
+      max_age_seconds = number      # (Optional) The time in seconds that your browser is to cache the preflight response for the specified resource.
+    }))
+  })
+  default = null
+}
+
+variable "intelligent_tiering_configuration" {
+  description = "Intelligent_tiering_configurations for the S3 bucket"
+  type = object({
+    name   = string        # (Required) The unique name used to identify the S3 Intelligent-Tiering configuration for the bucket.
+    status = string        # (Required) The status of the rule. Either "Enabled" or "Disabled". The rule is ignored if status is not "Enabled".
+    filter = object({      # (Optional) Filter that identifies subset of objects to which the replication rule applies
+      prefix = string      # (Optional) An object key name prefix that identifies the subset of objects to which the configuration applies.
+      tags   = map(string) # (Optional) All of these tags must exist in the object's tag set in order for the configuration to apply.
+    })
+    tiering = set(object({ # (Required) The S3 Intelligent-Tiering storage class tiers of the configuration
```

Comment:
Any particular reason for choosing `set()` here?

Reply:
Changed to list